src/main/java/org/broad/igv/track/TribbleFeatureSource.java

```diff
@@ -332,45 +333,30 @@ private NonIndexedFeatureSource(AbstractFeatureReader basicReader, FeatureCodec
             super(basicReader, codec, genome);
 
             featureMap = new HashMap<>(25);
-            Iterator<Feature> iter = null;
 
-            try {
-                iter = reader.iterator();
+            try (CloseableIterator<Feature> iter = reader.iterator()) {
                 while (iter.hasNext()) {
                     Feature f = iter.next();
                     if (f == null) continue;
 
                     String seqName = f.getChr();
                     String igvChr = genome == null ? seqName : genome.getCanonicalChrName(seqName);
 
-                    List<Feature> featureList = featureMap.get(igvChr);
-                    if (featureList == null) {
-                        featureList = new ArrayList();
-                        featureMap.put(igvChr, featureList);
-                    }
+                    List<Feature> featureList = featureMap.computeIfAbsent(igvChr, k -> new ArrayList<>());
```

Comment:
I did a bit of refactoring in this method, every where else is basically just renames.